core/src/main/java/hudson/model/labels/LabelExpression.java

```diff
@@ -340,27 +278,27 @@ public static FormValidation validate(@Nullable String expression, @CheckForNull
             Label.parseExpression(expression);
         } catch (ANTLRException e) {
             return FormValidation.error(e,
-                    Messages.AbstractProject_AssignedLabelString_InvalidBooleanExpression(e.getMessage()));
+                    Messages.LabelExpression_InvalidBooleanExpression(e.getMessage()));
         }
         final Jenkins j = Jenkins.get();
         Label l = j.getLabel(expression);
         if (l.isEmpty()) {
             for (LabelAtom a : l.listAtoms()) {
                 if (a.isEmpty()) {
                     LabelAtom nearest = LabelAtom.findNearest(a.getName());
-                    return FormValidation.warning(Messages.AbstractProject_AssignedLabelString_NoMatch_DidYouMean(a.getName(),nearest.getDisplayName()));
+                    return FormValidation.warning(Messages.LabelExpression_NoMatch_DidYouMean(a.getName(),nearest.getDisplayName()));
                 }
             }
-            return FormValidation.warning(Messages.AbstractProject_AssignedLabelString_NoMatch());
+            return FormValidation.warning(Messages.LabelExpression_NoMatch());
         }
         if (item != null) {
-            if (item instanceof AbstractProject) { // Use any project-oriented label validators
-                final AbstractProject<?, ?> project = (AbstractProject<?,?>) item;
-                for (AbstractProject.LabelValidator v : j.getExtensionList(AbstractProject.LabelValidator.class)) {
-                    FormValidation result = v.check(project, l);
-                    if (!FormValidation.Kind.OK.equals(result.kind)) {
-                        return result;
-                    }
+            // Use the project-oriented validators (including any that might implement checkItem to support non-Project
+            // items too).
+            // FIXME: Perhaps these should aggregate their errors/warnings?
```

Comment:
Unnecessary IMO; `AbstractProject.LabelValidator` should just be deprecated.

Reply:
Without applying `AbstractProject.LabelValidator` Instances, existing plugins implementing that extension point would stop being applied (and there would be no way for them to support pipeline jobs without changing their core dependency).